libraries/chain/webassembly/runtimes/eos-vm.cpp

```diff
@@ -476,6 +476,11 @@ REGISTER_CF_HOST_FUNCTION(action_data_size);
 REGISTER_CF_HOST_FUNCTION(current_receiver);
 REGISTER_HOST_FUNCTION(set_action_return_value);
 
+// sync call api
+REGISTER_HOST_FUNCTION(call);
+REGISTER_CF_HOST_FUNCTION(get_call_data);
+REGISTER_HOST_FUNCTION(set_call_return_value);
```

Comment:
Does it make sense for `get_call_data` to be allowed in CFA? If it isn't possible to `call` in a CFA can `get_call_data` ever do anything useful in a CFA?

Reply:
A CFA is designed so they can be removed. They should have no side-effects. They can only assert. If they are on the blockchain then they didn't assert, so you can just not run them. Having a `set_call_return_value` would mean that the trace would have to be different. Also would need to somehow store that for creating the action digest.

Reply:
Thanks for the explanation. I just thought a CFA cannot access (read/write) the state. 

For comparison, existing `read_action_data` can be used in CFA while `set_action_return_value` cannot.

Maybe I should change `get_call_data` to `read_call_data` to be consistent with `read_action_data`?

@arhag How do you think about sync calls in CFAs?

Reply:
I don't think we should allow sync calls in context-free actions.

Reply:
Thanks. I will keep all of them not to be called in context-free actions as is. Will add a comment there.